packages/d3fc-webgl/src/buffers/floatArrayFactory.js

```diff
@@ -0,0 +1,2 @@
+export default (data, requiredLength) =>
+    data.length >= requiredLength ? data : new Float32Array(requiredLength);
```

Comment:
This makes presumptions about `data`. I'd expect function to be a component which caches the previously requested array and optionally returns it if it's still useful.

Reply:
Think it's closer to that vision now.